tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -26,13 +26,19 @@ def get_resource_group(resource_id):
 
     @staticmethod
     def get_resource_type(resource_id):
-        return resource_id.split('/')[7]
+        return resource_id.split('/')[7].lower()
 
     @staticmethod
     def get_resource_name(resource_id):
         return resource_id.split('/')[8]
 
 
+def provider_resource_type_comparator(provider_resource_type, resource_id_resource_type):
+    """Compares a provider's resource type and a resource ids resource type
+    """
+    return provider_resource_type == resource_id_resource_type.split('/')[-1]
```

Comment:
If we are going to have this maybe make it foolproof?  Do an insensitive compare? `x.lower() == y.lower()` ?  Then you don't have to change get_resource_type and reduce chances of introducing bugs (or maybe if that introduces one we need to find it anyway.... since we know this can happen :) )

Reply:
Personally I don't know if I'd even look to see if this function existed anywhere if I had to implement the same thing since it's just a string comparison. 

When I was thinking about it more I thought it might be cleaner to add a method to get the resource type to the ParseId class, and then just have the string comparison where it was and add .lower() on both strings? I know we already get the resource type from the ID in the ParseID class - so what is the use case where we need the extra split again here?

Reply:
Agreed, that's a better implementation, I'll revert the `get_resource_type` and add the logic in here.